### PR TITLE
Feature/basic proxy

### PR DESCRIPTION
**Pull request by merwanehamadi**
### Background

Proxy to beat the challenges.
### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -17,5 +19,40 @@ def vcr_config():
...
+
+    def patched_init(requestor, *args, **kwargs):
+        original_init(requestor, *args, **kwargs)
+        patch_api_base(requestor)
+
+    def patched_validate_headers(self, supplied_headers):
+        headers = original_validate_headers(self, supplied_headers)
+        headers["AGENT-MODE"] = os.environ.get(
+            "AGENT_MODE",
+            "AAAAAAAAAAAAAAAAAAAAAMLheAAaAAAA0%2BuSeid%2BULvsea4JtiGRiSDSJSI%3DEUifiRBkKG5E2XzMDjRfl76ZC9Ub0wnz4XsNiRVBChTYbJcE3F",
```
Comment (dschonholtz) [inline tests/integration/conftest.py]: What's with the default value checked in here?
It looks like a secret of some kind?

File: tests/test_api_manager.py [hunk 2]
```diff
@@ -55,6 +56,7 @@ def test_create_chat_completion_empty_messages():
 
         with patch("openai.ChatCompletion.create") as mock_create:
             mock_response = MagicMock()
+            del mock_response.error
```
Comment (dschonholtz) [inline tests/test_api_manager.py:59]: Why are we deleting errors form the mock response?

**Comment by dschonholtz** [comment]
At a glance this looks good. I still need to pull it down and test some stuff locally and walk through it with the debugger. 
I also want to look at the corresponding proxy in the other repo. 
My aim is to actually have a full review done by EOD.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -74,6 +80,20 @@ jobs:
...
+          PROXY: ${{ vars.PROXY }}
+          AGENT_MODE: ${{ vars.AGENT_MODE }}
+          AGENT_TYPE: ${{ vars.AGENT_TYPE }}
 
       - name: Upload coverage reports to Codecov
         uses: codecov/codecov-action@v3
+
+      - name: Stage new files and commit
+        run: |
+          git add .
```
Comment (ntindle) [inline .github/workflows/ci.yml]: Specifically match files here

**Review by ntindle — CHANGES_REQUESTED** [review]
Adding . Could be a major pathway to commit any code to the repo

**Comment by dschonholtz** [comment]
Yeah, basic clean up and then we are good to go. 
I'm going to be running this locally just to make sure the agent works.
We also should actually fill out our PR checklist. 
And then I'll read through stuff one last time

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dschonholtz) [reply .github/workflows/ci.yml]: Yeah, just pulling in the cassette dirs would be better

**Comment by dschonholtz** [comment]
Agent runs locally fine on my machine. 
When we get the description updated and only commit cassettes on the new PR I think we are in business

**Comment by merwanehamadi** [comment]
@ntindle corrected thanks

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -3,12 +3,12 @@ name: Python CI
 on:
   push:
     branches: [ master ]
-  pull_request:
+  pull_request_target:
```
Comment (ntindle) [inline .github/workflows/ci.yml:6]: Would rather this run in a different workflow than changing the CI workflow

**Review by ntindle — COMMENTED** [review]
Concerned about changes to the CI workflow trigger. Imo it should trigger a separate workflow that is based on the branch

**Review by dschonholtz — DISMISSED** [review]
Approved. So we can start testing it!

**Review by richbeales — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (waynehamadi) [reply .github/workflows/ci.yml:6]: @ntindle  I thought of creating another test.yml, but then the ci.myl only had the lints and it was awkward. The lint doesn't need the pull_request_target but it's ok

**Pull request merged**